### DISCUSSION TITLE
Doing spectral reductions without using the image reduction

### DISCUSSION THREAD

**Discussion by rjs3273** [cylammarco/ASPIRED#111] [General]
In the past I just followed the recipes in the quickstart guide. They start with doing the basic image reductions. When I am starting with already reduced CCD images and just want to do the spectra reductions in ASPIRED, what is the best way of skipping over ImageReduction() and going straight to TwoDSpec()?

In the past I made an input filelist that had no bias, dark & flat and passed it through image_reduction.reduce(). That sort of does nothing, but it does initialise a whole load of ASPIRED presets and get everything organised in a format ready to pass into spectral_reduction. Something like:

  target_list.list text file:
    light, v_s_20230402_59_1_0_1.fits
    arc, v_a_20230402_60_1_0_1.fits 
  
```
targetRed = image_reduction.ImageReduction('target_list.list')
targetRed.reduce()
target2D = spectral_reduction.TwoDSpec(targetRed)
```

It seems easier to jump straight to spectral_reduction and never use image_reduction.

So, something like the following?...

```
datafits = fits.open('v_s_20230402_59_1_0_1.fits')
arcfits = fits.open('v_a_20230402_60_1_0_1.fits')
target2D = spectral_reduction.TwoDSpec()
target2D.add_data(datafits)
target2D.add_arc(arcfits)
target2D.set_exptime_keyword('EXPTIME')
target2D.set_gain(5.7)
target2D.set_airmass_keyword('AIRMASS')
target2D.set_properties(saxis=1, flip=False, cosmicray=False)
target2D.ap_trace(nspec=1, nwindow=25, fit_deg=3, display=True)  
```

And then from there on I can just follow the recipes in the quickstart guides again.

**Comment by cylammarco** [comment]
Yes, using `add_data()` is the way it is designed to start the processing without using an `ImageReduction` object. For the benefit of the other readers. I will elaborate briefly on `add_data()`.

`add_data()' takes two arguments: first is the data (compulsory), it can be of type ImageReduction, numpy array, CCDData, fits HDUList or fits ImageHDU; the second one is optional, it has to be of type `astropy.io.fits.header.Header`. If the header is provided, it will override the header (if available) from the data object provided (which is likely the case if you provide an ImageReduction object, fits HDUList or fits ImageHDU).

**Comment by cylammarco** [comment]
I will update the readthedocs before closing this.

**Comment by cylammarco** [comment]
For SPRAT data, I believe only `gain` is missing from the header.
